src/main/java/com/capitalone/dashboard/collector/DefaultWhiteSourceClient.java

```diff
@@ -0,0 +1,316 @@
+package com.capitalone.dashboard.collector;
+
+import com.capitalone.dashboard.client.RestClient;
+import com.capitalone.dashboard.model.LibraryPolicyResult;
+import com.capitalone.dashboard.model.LibraryPolicyThreatDisposition;
+import com.capitalone.dashboard.model.LibraryPolicyThreatLevel;
+import com.capitalone.dashboard.model.LibraryPolicyType;
+import com.capitalone.dashboard.model.WhiteSourceComponent;
+import com.capitalone.dashboard.model.WhiteSourceProduct;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class DefaultWhiteSourceClient implements WhiteSourceClient {
+    private static final Log LOG = LogFactory.getLog(DefaultWhiteSourceClient.class);
+    private static final String API_URL = "/api/v1.3";
+    private static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
+    private final RestClient restClient;
+    private final WhiteSourceSettings whiteSourceSettings;
+
+    @Autowired
+    public DefaultWhiteSourceClient(RestClient restClient, WhiteSourceSettings settings) {
+        this.restClient = restClient;
+        this.whiteSourceSettings = settings;
+    }
+
+    @Override
+    public String getOrgDetails(String instanceUrl, String orgToken) {
+        try {
+            JSONObject jsonObject = makeRestCall(getApiBaseUrl(instanceUrl), Constants.RequestType.getOrganizationDetails, orgToken, null, null);
+            String orgName = (String) jsonObject.get(Constants.ORG_NAME);
+            return orgName;
+        } catch (Exception e) {
+            LOG.error("Exception occurred while calling getOrgDetails " + e.getStackTrace());
+        }
```

Comment:
Should throw the exception out and catch it in the main loop. O/w may cause NPE

Reply:
changes made to throw exception and skip to next org